File: src/main/java/com/example/store/exception/BadRequestException.java
package com.example.store.exception;

public class BadRequestException extends RuntimeException {

    /**
     *
     */
    private static final long serialVersionUID = 1L;

    public final String message;

    public BadRequestException(String message) {
        super(message);
        this.message = message;
    }

    public BadRequestException(String message, Throwable t) {
        super(message, t);
        this.message = message;
    }
}

File: src/main/kotlin/com/example/store/exception/BadRequestException.kt
package com.example.store.exception

class BadRequestException : RuntimeException {
    override val message: String?

    constructor(message: String?) : super(message) {
        this.message = message
    }

    constructor(message: String?, t: Throwable?) : super(message, t) {
        this.message = message
    }

    companion object {
        /**
         *
         */
        private const val serialVersionUID = 1L
    }
}